impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/OWLLiteralImplBoolean.java

```diff
@@ -204,7 +209,21 @@ protected int compareObjectOfSameType(OWLObject object) {
         if (compareTo != 0) {
             return compareTo;
         }
-        return Boolean.valueOf(literal).compareTo(Boolean.valueOf(other.parseBoolean()));
+        return compareBoolean(literal, other.parseBoolean());
+    }
+
+    private static int compareBoolean(boolean a, boolean b) {
+        if (a) {
+            if (b) {
+                return 0;
+            } else {
+                return 1;
+            }
+        } else if (!b) {
+            return 0;
```

Comment:
This case looks a little strange. If a and b are reversed it will return -1 from the case below, which doesn't fulfill the general Comparable contract.

In the past I have used something like the following from memory:

    if (a!=b) {
        if (b) {
            return 1;
        } else {
            return -1;
    } else {
        return 0;
    }